internal/core/data/handler.go

```diff
@@ -0,0 +1,186 @@
+	newId, err := persistEventToDb(event)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	go putEventOnQueue(event, ctx)
+	w.WriteHeader(http.StatusOK)
+	w.Write([]byte(newId))
+}
+
+// UpdateEvent update existing event data for requests with data represented in JSON format
+func (jeh JsonEventHandler) UpdateEvent(w http.ResponseWriter, req *http.Request) {
+	updateEventAndPersist(w, req)
+}
+
+// CreateNewEvent persists event data for requests with event represented in CBOR format
+func (CborEventHandler) CreateNewEvent(w http.ResponseWriter, req *http.Request) {
+	bodyReader, err := NewReader(req)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	bytes, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		LoggingClient.Error("Error reading bytes from request body: " + err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	ctx := req.Context()
+	var event models.Event
+	err = bodyReader.Read(bytes, &ctx, &event)
+	// Problem Decoding Event
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	err = validateEvent(event, ctx)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	LoggingClient.Info("Posting Event: " + event.String())
+	id, err := PersistAndPublish(event, ctx)
+	if err != nil {
+		LoggingClient.Error("Error persisting and/or publishing event: " + err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusAccepted)
+	w.Write([]byte(id))
+}
+
+// UpdateEvent update existing event data for requests with data represented in CBOR format
+func (CborEventHandler) UpdateEvent(w http.ResponseWriter, req *http.Request) {
+	updateEventAndPersist(w, req)
+}
+
+func updateEventAndPersist(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	bodyReader, err := NewReader(req)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	var from models.Event
+	bytes, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		LoggingClient.Error("Error reading bytes from request body: " + err.Error())
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	err = bodyReader.Read(bytes, &ctx, &from)
+	// Problem Decoding Event
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	LoggingClient.Info("Updating event: " + from.ID)
+	err = updateEvent(from, ctx)
+	if err != nil {
+		switch t := err.(type) {
+		case *ierror.ErrEventNotFound:
+			http.Error(w, t.Error(), http.StatusNotFound)
+		default:
+			http.Error(w, t.Error(), http.StatusInternalServerError)
+		}
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write([]byte("true"))
+}
+
+func publishEventData(checksum string, event models.Event, ctx context.Context) {
```

Comment:
Just an observation regarding API consistency.. 
I see you've introduced two methods for persisting an event with/without checksum **persistEventToDb** and **persistEventToDbWithChecksum** 
However, the **publishEventData** method is overloaded to publish event data with and without checksum, requiring caller to supply empty string if not in scenario where they'll apply checksum.

Reply:
For the case where there is no checksum information I was intending on using the [putEventOnQueue](https://github.com/edgexfoundry/edgex-go/blob/d8b36499b9ebed8c04bf5a46aea7dce45159b10f/internal/core/data/event.go#L245) function which provides the functionality of `publishEventData` but without providing the checksum information. For consistency I can rename `publishEventData` to `publishEventDataWithChecksum` so it is more clear to the reader.

Reply:
Resolving conversation as this logic has been removed entirely.